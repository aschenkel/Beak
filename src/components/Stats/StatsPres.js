import React, { Component } from 'react';
import {
  View,
  Text,
  StyleSheet,
  FlatList
} from 'react-native';
import StatCard from '../TwitCard';
class StatsPres extends Component{

    render(){
        return (
            <View>
                 <FlatList
                    scrollEnabled={ false }
                    data={this.props.topTweets}
                    keyExtractor={this._keyExtractor}
                    renderItem={this._renderItem}
                />    
            </View>   
               
            )
    }
    _keyExtractor = (item, index) => item.id;

    _renderItem = ({item,index}) => (
        <TwitCard 
            id={item.id}
            text={item.text}
            avgGrade={this.props.avgGrade}
            grade={item.grade}
            favCount={item.favorite_count}
            created_at={item.created_at}
        />
  );

}

export default StatsPres