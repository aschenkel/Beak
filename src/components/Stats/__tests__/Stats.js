import React from 'react';
import StatsPres from '../StatsPres';
import renderer from 'react-test-renderer';

it('renders user stats', () => {
  var topTweetsData= {
                tweets:
                    [
                        {text:"first", 
                            user:{
                                name:"test",screen_name:"tester"
                            }
                        },
                        {text:"second tweet",
                            user:{
                                name:"test",screen_name:"tester"
                            }
                        },
                    ],
                grades:[2,4]
            }
  const tree = renderer.create(
    <StatsPres avgGrade={3} 
            topTweets={topTweetsData}
        />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
