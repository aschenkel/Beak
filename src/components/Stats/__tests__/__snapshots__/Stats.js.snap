// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user stats 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 16,
      }
    }
  >
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
        Average: 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      3
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        th
      </Text>
       grade
    </Text>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "text": "first",
          "user": Object {
            "name": "test",
            "screen_name": "tester",
          },
        },
        Object {
          "text": "second tweet",
          "user": Object {
            "name": "test",
            "screen_name": "tester",
          },
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={undefined}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F5FCFF",
                "flex": 1,
                "margin": 5,
              },
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0.3,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "margin": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 2,
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  test
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "grey",
                      "marginLeft": 4,
                    }
                  }
                >
                  @
                  tester
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "red",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "red",
                      "fontSize": 9,
                    }
                  }
                >
                  unsmartest
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "marginTop": 4,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              >
                first
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 15,
                }
              }
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "grey",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        undefined,
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 16,
                    "marginLeft": 8,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 48,
                  }
                }
              >
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "grey",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 16,
                      "marginLeft": 8,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "marginLeft": 80,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "gold",
                      "fontSize": 20,
                      "fontWeight": "bold",
                    }
                  }
                >
                  2
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={undefined}
                  >
                    nd
                  </Text>
                   grade
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F5FCFF",
                "flex": 1,
                "margin": 5,
              },
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0.3,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "margin": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 2,
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  test
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "grey",
                      "marginLeft": 4,
                    }
                  }
                >
                  @
                  tester
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "blue",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "blue",
                      "fontSize": 9,
                    }
                  }
                >
                  smartest
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "marginTop": 4,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              >
                second tweet
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 15,
                }
              }
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "grey",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        undefined,
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 16,
                    "marginLeft": 8,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 48,
                  }
                }
              >
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "grey",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 16,
                      "marginLeft": 8,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "marginLeft": 80,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "gold",
                      "fontSize": 20,
                      "fontWeight": "bold",
                    }
                  }
                >
                  4
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={undefined}
                  >
                    th
                  </Text>
                   grade
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
